### DISCUSSION TITLE
Any chance  this can work on a browser?

### DISCUSSION THREAD

**Discussion by MindChamber** [ikemen-engine/Ikemen-GO#1627] [Feature Request]
### Is your feature request related to a problem? Please describe.

I like to upload games  my games on Newgrounds.

### Describe the solution you'd like

any chance this is being worked on?  or if possible maybe chip in towards the funds to make this happen?

**Comment by formula1** [comment]
Hypothetically, it can build to wasm as well as have javascript bindings to javascript technology. 
- Input - https://developer.mozilla.org/en-US/docs/Web/API/Gamepad_API/Using_the_Gamepad_API
- FileSystem - https://www.npmjs.com/package/browserfs
  - Use Zip File as base game (characters, stages)
  - Create a LocalStorage for user configuration
  - Use Memory For storage of storing unneeded file
- Audio - https://developer.mozilla.org/en-US/docs/Web/API/AudioContext
- Display - https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API


https://github.com/ikemen-engine/Ikemen-GO/discussions/879

I tried to set up ikemen go to use generic functions/classes so that we can use the same api for both native and wasm version (https://github.com/formula1/ikemen-go-js-bindings/blob/master/native/file_abstract.go)
It's relatively easy for Go to interact with javascript globals (https://github.com/formula1/ikemen-go-js-bindings/blob/master/go/src/testing.go)

**Reply by james-pre** [reply]
Hi @formula1, I'm the maintainer of BrowserFS. I'd like to let you know that BrowserFS has been deprecated. You can read more about that on the BrowserFS readme: https://github.com/jvilk/BrowserFS. [ZenFS](https://github.com/zen-fs/core) is an updated and active fork that I created.

**Comment by SuperFromND** [comment]
We've actually had some partial work for browser support done by @samhocevar, although it isn't particularly well-maintained nor was performance particularly great if I recall. :<
